internal/internal_task_handlers.go

```diff
@@ -653,7 +655,26 @@ ProcessEvents:
 		// check if decisions from reply matches to the history events
 		if err := matchReplayWithHistory(replayDecisions, respondEvents); err != nil {
```

Comment:
I think there is another place where non-deterministic error could happen, which is in the state machine transition. I think we need to handle that as well.

Reply:
As discussed, this will be fixed separately as there's some unknown corner cases.